### PR TITLE
fix(form-field): scrollbars appear on autosize textarea in chrome

### LINKED ISSUE #10376: Textarea with matTextareaAutosize has spinner arrows
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
textarea element with matTextareaAutosize should not have spinner arrows

#### What is the current behavior?
textarea element with matTextareaAutosize has spinner arrows that go away after entering the first character

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-talhyo

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Chrome Version 65.0.3325.146

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #10376

### CODE REVIEW

File: src/cdk/text-field/_text-field.scss [hunk 1]
```diff
@@ -19,6 +19,13 @@
   textarea.cdk-textarea-autosize {
     resize: none;
   }
+
+  textarea.cdk-textarea-autosize-measuring {
+    height: auto !important;
```
Comment (jelbourn) [inline src/cdk/text-field/_text-field.scss]: Add comments that explains what this class is for and the use of `!important`?

File: src/cdk/text-field/_text-field.scss [hunk 2]
```diff
@@ -19,6 +19,13 @@
...
+    overflow: hidden !important;
+    padding: 2px 0 !important;
```
Comment (jelbourn) [inline src/cdk/text-field/_text-field.scss]: Explain the origin of the `2px` padding?

File: src/cdk/text-field/autosize.ts [hunk 3]
```diff
@@ -187,13 +187,15 @@ export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
...
-    textarea.style.overflow = 'hidden';
+    textarea.classList.add('cdk-textarea-autosize-measuring');
     textarea.placeholder = '';
 
     // Use the scrollHeight to know how large the textarea *would* be if fit its entire value.
-    textarea.style.height = `${textarea.scrollHeight}px`;
-    textarea.style.overflow = '';
+    // We subtract 4 from the scrollHeight because of the 2px top and bottom padding. Having the
+    // extra 2px on either side seems to fix inaccurate measurements in Chrome.
+    textarea.style.height = `${textarea.scrollHeight - 4}px`;
```
Comment (jelbourn) [inline src/cdk/text-field/autosize.ts]: Phrase something like... "The cdk-textarea-autosize-measuring class includes a 2px padding to workaround an issue with Chrome, so we account for that extra space here by subtracting 4 (2px top + 2px bottom)."

File: src/lib/input/input.scss [hunk 4]
```diff
@@ -92,3 +92,8 @@ textarea.mat-input-element {
     resize: none;
   }
 }
+
+textarea.mat-input-element {
+  padding: 2px 0;
```
Comment (jelbourn) [inline src/lib/input/input.scss]: Add comment here explaining this as well?

**Comment by mmalerba** [comment]
Looks like a flake, re-ran it and it went away

**Pull request merged**